radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -57,70 +64,100 @@ public EventCoordinator(
 		SafetyRules safetyRules,
 		Pacemaker pacemaker,
 		VertexStore vertexStore,
+		PendingVotes pendingVotes,
 		ProposerElection proposerElection,
-		@Named("self") EUID self
+		QuorumRequirements quorumRequirements,
+		@Named("self") ECKeyPair selfKey
 	) {
 		this.proposalGenerator = Objects.requireNonNull(proposalGenerator);
 		this.mempool = Objects.requireNonNull(mempool);
 		this.networkSender = Objects.requireNonNull(networkSender);
 		this.safetyRules = Objects.requireNonNull(safetyRules);
 		this.pacemaker = Objects.requireNonNull(pacemaker);
 		this.vertexStore = Objects.requireNonNull(vertexStore);
+		this.pendingVotes = Objects.requireNonNull(pendingVotes);
 		this.proposerElection = Objects.requireNonNull(proposerElection);
-		this.self = Objects.requireNonNull(self);
+		this.quorumRequirements = Objects.requireNonNull(quorumRequirements);
+		this.selfKey = Objects.requireNonNull(selfKey);
 	}
 
-	private void processNewRound(Round round) {
-		log.debug("Processing new round: " +  round);
+	private void processNewView(View view) {
+		log.debug("Processing new view: " + view);
 		// only do something if we're actually the leader
-		if (!proposerElection.isValidProposer(self, round)) {
+		if (!proposerElection.isValidProposer(selfKey.getUID(), view)) {
 			return;
 		}
 
-		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentRound());
+		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentView());
 
 		// TODO: Handle empty proposals
 		if (proposal.getAtom() != null) {
 			this.networkSender.broadcastProposal(proposal);
+			log.info("Starting view " + view + " with proposal " + proposal);
 		}
 	}
 
 	public void processVote(Vote vote) {
-		// only do something if we're actually the leader for the next round
-		if (!proposerElection.isValidProposer(self, vote.getVertexMetadata().getRound().next())) {
-			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getRound()));
+		// only do something if we're actually the leader for the next view
+		if (!proposerElection.isValidProposer(selfKey.getUID(), vote.getVertexMetadata().getView().next())) {
+			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getView()));
 			return;
 		}
 
-		// accumulate votes into QCs
-		// TODO assumes a single node network for now
-		QuorumCertificate qc = new QuorumCertificate(vote, vote.getVertexMetadata());
-		this.safetyRules.process(qc);
+		// accumulate votes into QCs in store
+		Optional<QuorumCertificate> potentialQc = this.pendingVotes.insertVote(vote, this.quorumRequirements);
```

Comment:
quorumRequirements should be a depdendency of PendingVotes no?

Reply:
It seemed more flexible to have it as an argument, but you're right, making it a dependency for now is simpler.